rotkehlchen/tests/unit/uniswap/conftest.py

```diff
@@ -1,19 +1,35 @@
-from unittest.mock import MagicMock, patch
+from unittest.mock import patch
 
 import pytest
 
 from rotkehlchen.chain.ethereum.uniswap.uniswap import Uniswap
 from rotkehlchen.premium.premium import Premium
 
 
+@pytest.fixture(name='mock_graph')
+def fixture_graph():
+    with patch('rotkehlchen.chain.ethereum.uniswap.uniswap.Graph'):
```

Comment:
interesting. So with using this `with` and `yield` from a fixture you essentially can consider the entire test under the context manager, right?

Reply:
yes as far as I know. I wasn't sure if you would like this pattern and I considered different options:
1. Somehow have a general fixture for Graph and be confident wherever is used `self.graph` will be a MagicMock (but his could affect other DeFi modules tests).
2. Just patch inside `mock_uniswap`.
3. Patch only on those tests that call `self.graph.query`.

I finally opted for implementing it as a fixture only for uniswap tests.

Reply:
I generally like the simple patching and abstracting in a function to reuse. This allows it to be used only in parts of the test if required.

But your solution is great if you want to use the mocking in an entire test. And since this is how you use it it's great.

Reply:
And the added bonus is that since you use it in all of the test, the test code becomes more readable since it does not need to be in a context manager.

May use this pattern myself in the future for a similar situation.